shm/Player.cpp

```diff
@@ -0,0 +1,36 @@
+#include "Player.h"
+
+#include <numeric>
+
+Player::Player(Ship ship, size_t money)
+    : ship_(std::make_unique<Ship>(ship)), money_(money) {}
+
+size_t Player::getAvailableSpace() const {
+    return available_space_;
+}
+
+size_t Player::getMoney() const {
+    return money_;
+}
+
+Ship* Player::getShip() const {
+    return ship_.get();
+}
+
+size_t Player::getSpeed() const {
+    return ship_->getSpeed();
+}
+
+Cargo* Player::getCargo(size_t index) const {
+    if (ship_) {
+        return ship_->getCargo(index).get();
```

Comment:
What if index is out of range? You got nullptr? If yes, you will call get() on uninitialized unique_ptr and got crash.